### PR TITLE
adjust spring usage from field injection to constructor injection

### PR DESCRIPTION
**Pull request by dasniko**
Proposal, like discussed in https://github.com/togglz/togglz/issues/261#issuecomment-349565115
Please review and merge, if ok.

### CODE REVIEW

File: documentation/spring-boot-starter.html.haml [hunk 1]
```diff
@@ -71,21 +71,29 @@ layout: base
...
-    @RequestMapping("/")
-    public ResponseEntity<?> index() {
-        if (manager.isActive(HELLO_WORLD)) {
-             ...
-        }
+    @Controller
+    public class MyClass {
+      private FeatureManager manager;
+
+      public MyClass(Feature manager) {
```
Comment (chkal) [inline documentation/spring-boot-starter.html.haml]: I guess `Feature` should be `FeatureManager`?

Comment (dasniko) [reply documentation/spring-boot-starter.html.haml]: Double-checked it several times to not run into a typo and what happens? A typo... 😞

Comment (chkal) [reply documentation/spring-boot-starter.html.haml]: :laughing:

**Comment by chkal** [comment]
Merged! Thanks a lot!

**Pull request merged**